src/averageAges.js

```diff
@@ -37,8 +43,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const filteredWomen = withChildren
+    ? people.filter(
+      person => people.some(child => child.mother === person.name)
+      && person.sex === 'f')
+    : people.filter(person => person.sex === 'f');
+
+  const totalAge = filteredWomen.reduce(
+    (sum, person) => sum + (person.died - person.born), 0);
+  const averageAge = totalAge / filteredWomen.length;
+
+  return averageAge;
```

Comment:
Same as in `calculateMenAverageAge`.
You have `getAverageAge` helper. Why you didn't use it here?